packages/@aws-containers/aws-ecs-builder/lib/extensions/appmesh.ts

```diff
@@ -0,0 +1,276 @@
+    } else {
+      this.protocol = appmesh.Protocol.HTTP;
+    }
+  }
+
+  public prehook(service: Service, scope: cdk.Construct) {
+    this.parentService = service;
+    this.scope = scope;
+
+    // Make sure that the parent cluster for this service has
+    // a namespace attached.
+    if (!this.parentService.cluster.defaultCloudMapNamespace) {
+      this.parentService.cluster.addDefaultCloudMapNamespace({
+        name: 'internal',
+      });
+    }
+  }
+
+  public mutateTaskDefinitionProps(props: ecs.TaskDefinitionProps) {
+    // Find the app extension, to get its port
+    const containerextension = this.parentService.serviceDescription.get('service-container') as Container;
+
+    if (!containerextension) {
+      throw new Error('Appmesh extension requires an application extension');
+    }
+
+    return {
+      ...props,
+
+      // App Mesh requires AWS VPC networking mode so that each
+      // task can have its own IP address
+      networkMode: ecs.NetworkMode.AWS_VPC,
+
+      // This configures the envoy container as a proxy for all
+      // traffic going into and out of the task, with a few exceptions
+      // for metadata endpoints or other ports that need direct
+      // communication
+      proxyConfiguration: new ecs.AppMeshProxyConfiguration({
+        containerName: 'envoy',
+        properties: {
+          appPorts: [containerextension.trafficPort],
+          proxyEgressPort: 15001,
+          proxyIngressPort: 15000,
+          ignoredUID: 1337,
+          ignoredGID: 1338,
+          egressIgnoredIPs: [
+            '169.254.170.2', // Allow services to talk directly to ECS metadata endpoints
+            '169.254.169.254', // and EC2 instance endpoint
+          ],
+          // Note that at some point we will need other extensions like
+          // MySQL to be able to add their ports to this egress ignored
+          // ports automatically.
+          egressIgnoredPorts: [
+          ],
+        },
+      }),
+    } as ecs.TaskDefinitionProps;
+  }
+
+  public useTaskDefinition(taskDefinition: ecs.TaskDefinition) {
+    var region = cdk.Stack.of(this.scope).region;
+    var appMeshRepo;
+    var ownerAccount;
+
+    // This is currently necessary because App Mesh has different images in each region,
+    // and some regions have their images in a different account. See:
+    // https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html
+    if (region === 'me-south-1') {
+      ownerAccount = 772975370895;
+    } else if (region === 'ap-east-1') {
+      ownerAccount = 856666278305;
+    } else {
+      ownerAccount = 840364872350;
+    }
+
+    appMeshRepo = ecr.Repository.fromRepositoryArn(
+      this.scope,
+      `${this.parentService.id}-envoy-repo`,
+      `arn:aws:ecr:us-east-1:${ownerAccount}:repository/aws-appmesh-envoy`,
+    );
+
+    this.container = taskDefinition.addContainer('envoy', {
+      image: ecs.ContainerImage.fromEcrRepository(appMeshRepo, 'v1.15.0.0-prod'),
```

Comment:
is this the canonical version that has to be used? I imagine this may change in the future, so again worth constantizing?